### DISCUSSION TITLE
Add option to encrypt public key

### DISCUSSION THREAD

**Discussion by pcarinhas** [FiloSottile/age#598] [General]
If one wanted to keep their public key protected by encrypting it by itself (It wouldn't really be a public key then of course), it would offer a more secure way to avoid spoofing of messages by a 3rd party, and other security issues. 

To use the public key, you'd have to decrypt it with the private key, of course. That could be done inside Age/Rage without exposing it.

Of course once can already do this either manually or by created some wrappers around Age/Rage.
If there is any merit to this, please discuss.

**Comment by pcarinhas** [comment]
Use case:

You want to ensure your backups are not spoofed by someone who may have your public key.
In such a case, an attacker could spoof a backup with trojan malware, then if the orginal systems are
destroyed, the owner then would restore malicious data to their environment.

In this scenario, nobody but the author has the public (and private) keys, so there is no way to spoof anything.

**Reply by awnumar** [reply]
I think your question has been answered [here](https://github.com/FiloSottile/age/discussions/613).
